site/AnnotatingDataValue.tsx

```diff
@@ -0,0 +1,78 @@
+import { faChartLine } from "@fortawesome/free-solid-svg-icons/faChartLine"
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
+import React, { useEffect, useState } from "react"
+import ReactDOM from "react-dom"
+import { Grapher } from "../grapher/core/Grapher"
+import { ENV } from "../settings/clientSettings"
+import { DataValue, DataValueProps, processTemplate } from "./DataValue"
+import { SiteAnalytics } from "./SiteAnalytics"
+
+const AnnotatingDataValue_name = "AnnotatingDataValue"
+const analytics = new SiteAnalytics(ENV)
+
+export const AnnotatingDataValue = ({
+    dataValueProps,
+    grapherInstance,
+}: {
+    dataValueProps: DataValueProps
+    grapherInstance?: Grapher
+}) => {
+    const [isInteractive, setInteractive] = useState(false)
+    const [label] = useState(processTemplate(dataValueProps))
+
+    const renderAnnotationInGrapher = () => {
+        grapherInstance?.renderAnnotation({
+            entityName: dataValueProps.entityName,
+            year: Number(dataValueProps.year),
+        })
+        analytics.logDataValueAnnotate(label)
+    }
+
+    useEffect(() => {
+        setInteractive(true)
+    }, [])
+
+    return (
+        <span className="annotating-data-value">
+            <script
+                data-type={AnnotatingDataValue_name}
+                type="component/props"
+                dangerouslySetInnerHTML={{
+                    __html: JSON.stringify(dataValueProps),
+                }}
+            ></script>
+            <span
+                onMouseEnter={renderAnnotationInGrapher}
+                onMouseLeave={grapherInstance?.resetAnnotation}
+                className={isInteractive ? "interactive" : ""}
+            >
+                <DataValue label={label}></DataValue>
+                {isInteractive ? <FontAwesomeIcon icon={faChartLine} /> : null}
+            </span>
+        </span>
+    )
+}
+
+export function runAnnotatingDataValue(
```

Comment:
Maybe rename to `hydrateAnnotatingDataValue`? That's the only thing it does right?